apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -12,50 +16,45 @@ import { ListItemComponent } from '../list-item/list-item.component';
     <div
       class="flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4"
       [class]="customClass">
-      <img
-        *ngIf="type === CardType.TEACHER"
-        src="assets/img/teacher.png"
-        width="200px" />
-      <img
-        *ngIf="type === CardType.STUDENT"
-        src="assets/img/student.webp"
-        width="200px" />
-
+      <ng-content select="img"></ng-content>
       <section>
-        <app-list-item
-          *ngFor="let item of list"
-          [name]="item.firstName"
-          [id]="item.id"
-          [type]="type"></app-list-item>
+        <ng-container *ngFor="let item of list">
+          <ng-template
+            [ngTemplateOutlet]="rowTemplate"
+            [ngTemplateOutletContext]="{ $implicit: item }"></ng-template>
+        </ng-container>
       </section>
 
       <button
         class="rounded-sm border border-blue-500 bg-blue-300 p-2"
-        (click)="addNewItem()">
+        (click)="add.emit()">
         Add
       </button>
     </div>
   `,
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  styles: [
+    `
+      :host {
+        .bg-light-red {
+          background-color: rgba(250, 0, 0, 0.1);
+        }
+        .bg-light-green {
+          background-color: rgba(250, 0, 0, 0.1);
+        }
+      }
+    `,
+  ],
+  imports: [NgFor, ListItemComponent, NgTemplateOutlet],
 })
 export class CardComponent {
   @Input() list: any[] | null = null;
```

Comment:
We should'nt use `any[]` type right here, type `any` is kind of "i dont care about the type" - that destroys the concept of **type**script.

Reply:
Hi @svenson95 ,

 Students component sent Students Array list under the list property. Same with teachers and city card components as well. I don't want my card component to have typing as **Student | Card | Teacher**
 
  I want to avoid the scenario of adding new types inside card component every time there is a new  implementation layer added for card component into the application.
  
  How to add this scenario ? can we have dynamic typing for better type casting ?

Reply:
Yeah it's possible to use a generic `T` type instead of `any`, so you can pass different types to this `list` input. Furthermore you can take advantage of `ngTemplateContextGuard`, but this is part of another challenge - [Typed ContextOutlet](https://angular-challenges-620x3ppdp-tomalaforges-projects.vercel.app/challenges/angular/4-context-outlet-typed/). I'd recommend you to checkout the community & author solutions of this challenge as well, just to see different ways of solving this problem and maybe learn something new 🚀